src/common/tracer.cc

```diff
@@ -55,6 +65,50 @@ bool Tracer::is_enabled() const {
   return g_ceph_context->_conf->jaeger_tracing_enable;
 }
 
+void encode(const jspan_context& span_ctx, bufferlist& bl, uint64_t f) {
+  if (span_ctx.IsValid()) {
+    using namespace opentelemetry;
+    using namespace trace;
+    const size_t ctx_size = TraceId::kSize + SpanId::kSize + 1;
+    char trace_identity[ctx_size];
+    uint8_t* trace_identity_ptr = (uint8_t*)trace_identity;
+    nostd::span<uint8_t, TraceId::kSize> bdika{trace_identity_ptr, TraceId::kSize};
+    span_ctx.trace_id().CopyBytesTo(bdika);
+    trace_identity_ptr += TraceId::kSize;
+    span_ctx.span_id().CopyBytesTo(nostd::span<uint8_t, SpanId::kSize>{trace_identity_ptr, SpanId::kSize});
+    trace_identity_ptr += SpanId::kSize;
+    span_ctx.trace_flags().CopyBytesTo(nostd::span<uint8_t, 1>{trace_identity_ptr, 1});
+
+    ENCODE_START(1, 1, bl);
+    encode(std::string_view(trace_identity, ctx_size), bl);
```

Comment:
you can probably append the two buffers to the bufferlist directly, instead of copying to the trace_identity buffer and then copy into the bufferlist

Reply:
does not matter if we don't use a buffer